### PR TITLE
corehost: align artifacts dir used by managed and native build.

### PR DESCRIPTION
**Pull request by tmds**
Fixes build failure that occurs when OutputRid and __DistroRid are not the same.

cc @omajid

### CODE REVIEW

File: src/native/corehost/build.sh [hunk 1]
```diff
@@ -74,14 +79,15 @@ __LogsDir="$__RootBinDir/log"
...
+__PackageRid="${__PackageRid:-$__DistroRid}"
 __DistroRidLower="$(echo $__DistroRid | tr '[:upper:]' '[:lower:]')"
 __BinDir="$__RootBinDir/bin/$__DistroRidLower.$__BuildType"
 __IntermediatesDir="$__RootBinDir/obj/$__DistroRidLower.$__BuildType"
 
 export __BinDir __IntermediatesDir __RuntimeFlavor
 
 __CMakeArgs="-DCLI_CMAKE_HOST_VER=\"$__host_ver\" -DCLI_CMAKE_COMMON_HOST_VER=\"$__apphost_ver\" -DCLI_CMAKE_HOST_FXR_VER=\"$__fxr_ver\" $__CMakeArgs"
-__CMakeArgs="-DCLI_CMAKE_HOST_POLICY_VER=\"$__policy_ver\" -DCLI_CMAKE_PKG_RID=\"$__DistroRid\" -DCLI_CMAKE_COMMIT_HASH=\"$__commit_hash\" $__CMakeArgs"
+__CMakeArgs="-DCLI_CMAKE_HOST_POLICY_VER=\"$__policy_ver\" -DCLI_CMAKE_PKG_RID=\"$__PackageRid\" -DCLI_CMAKE_COMMIT_HASH=\"$__commit_hash\" $__CMakeArgs"
```
Comment (tmds) [inline src/native/corehost/build.sh:90]: While changing this file, I noticed this use of __DistroRid. From the name of the property, I assume we want to pass in the PackageRid.

File: Directory.Build.props [hunk 2]
```diff
@@ -231,7 +231,7 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <DotNetHostBinDir>$([MSBuild]::NormalizeDirectory('$(ArtifactsBinDir)', '$(OutputRid).$(Configuration)', 'corehost'))</DotNetHostBinDir>
+    <DotNetHostBinDir>$([MSBuild]::NormalizeDirectory('$(ArtifactsBinDir)', '$(__DistroRid.ToLowerInvariant()).$(Configuration)', 'corehost'))</DotNetHostBinDir>
```
Comment (tmds) [inline Directory.Build.props:234]: This now matches with the logic in the corehost/build.sh script.

**Comment by am11** [comment]
> Fixes build failure that occurs when OutputRid and __DistroRid are not the same.

Do you mean "Add support for .." or is it an existing scenario which was working before and broke by some other change?

**Comment by tmds** [comment]
> Do you mean "Add support for .." or is it an existing scenario which was working before and broke by some other change?

I'm not aware this was working before. I think we're the first to try set `TargetRid` to something other than distro rid and find out the corehost artifacts path gets calculated differently in these files.

**Comment by tmds** [comment]
@am11 I've created an issue: https://github.com/dotnet/runtime/issues/74577. https://github.com/dotnet/runtime/pull/74504 supersedes this PR.

**Pull request closed without merging**